google/cloud/internal/tracing_rest_client.cc

```diff
@@ -25,6 +29,35 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
 #ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
 
+namespace {
+
+// We always inject the CloudTraceContext, these requests are going to GCP where
+// the X-Cloud-Trace-Context header can connect the request with the internal
+// tracing service.
```

Comment:
s/, these/. These/

---

I think this is like 99% true? I am thinking of WIF. Don't we use the `RestClient` to send requests to AWS, for example, in that case?

We can deal with that later, but in the long run, I suspect we want to have a propagator determined by Options. (If the Options know whether we are also communicating with AWS, Ideally we could use their X-Ray context too).

With that being said, adding this header in a request to some other cloud is fine. It will just get ignored.

Reply:
You are right, changed the comment and I opened the bug you wanted.